curp/src/server.rs

```diff
@@ -205,6 +206,61 @@ impl CmdBoardValue {
     }
 }
 
+/// The speculative pool that stores commands that might be executed speculatively
+#[derive(Debug)]
+pub(crate) struct SpeculativePool<C> {
+    /// Store
+    pub(crate) pool: VecDeque<C>,
+    /// Store the ids of commands that have completed execution and should be inserted into the speculative pool again
```

Comment:
Store the ids of commands that have completed backend syncing, but not in the local speculative pool.